iothub/device/tests/Mqtt/MqttTransportHandlerTests.cs

```diff
@@ -369,7 +370,13 @@ public async Task MqttTransportHandlerSendTwinGetAsyncHappyPath()
                 .Returns(msg =>
                 {
                     var response = new Message(twinByteStream);
-                    response.MqttTopicName = GetResponseTopic(msg.Arg<Message>().MqttTopicName, statusSuccess);
+                    string mqttTopic = GetResponseTopic(msg.Arg<Message>().MqttTopicName, statusSuccess);
+                    response.MqttTopicName = mqttTopic;
+                    var publishPacket = new PublishPacket(QualityOfService.AtMostOnce, false, false)
+                    {
+                        TopicName = mqttTopic
+                    };
+                    MqttIotHubAdapter.PopulateMessagePropertiesFromPacket(response, publishPacket);
                     transport.OnMessageReceived(response);
```

Comment:
the transport layer receives an mqtt packet and then converts it to a Message. The tests were omitting the mqtt packet step, so it has been added back in.

(The reason this was working even when the mqtt packet step was omitted was because we were extracting the same information from the mqtt topic again in the transport layer: https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs#L1101)